EssentialFeed/EssentialFeed/Feed API/RemoteFeedCommentsLoader.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Created by Azamat Valitov on 13.03.2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public class RemoteFeedCommentsLoader: FeedCommentsLoader {
+	
+	public enum Error: Swift.Error {
+		case connectivity
+		case invalidData
+	}
+	
+	private let url: URL
+	private let client: HTTPClient
+	
+	public init(url: URL, client: HTTPClient) {
+		self.url = url
+		self.client = client
+	}
+	
+	private final class HTTPClientTaskWrapper: FeedCommentsLoaderTask {
+		private var completion: ((FeedCommentsLoader.Result) -> Void)?
+		
+		var wrapped: HTTPClientTask?
+		
+		init(_ completion: @escaping (FeedCommentsLoader.Result) -> Void) {
+			self.completion = completion
+		}
+		
+		func complete(with result: FeedCommentsLoader.Result) {
+			completion?(result)
+		}
+		
+		func cancel() {
+			preventFurtherCompletions()
+			wrapped?.cancel()
+		}
+		
+		private func preventFurtherCompletions() {
+			completion = nil
+		}
+	}
+	
+	public func load(completion: @escaping (FeedCommentsLoader.Result) -> Void) -> FeedCommentsLoaderTask {
+		let task = HTTPClientTaskWrapper(completion)
+		task.wrapped = client.get(from: url, completion: {[weak self] result in
+			guard let self = self else { return }
+			self.handle(result, task)
```

Comment:
Using `self?.` you avoid the need for the guard. It can be a single line:

```suggestion
			self?.handle(result, task)
```